imports/ui/dialogs/unit-select-dialog.jsx

```diff
@@ -0,0 +1,160 @@
+import Units, { collectionName } from '../../api/units'
+import TextField from 'material-ui/TextField'
+import { NoItemMsg } from '../explorer-components/no-item-msg'
+import {
+  customTitleStyle,
+  closeDialogButtonStyle
+} from './generic-dialog.mui-styles'
+
+class UnitSelectDialog extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      searchMode: false,
+      searchText: '',
+      searchResult: []
+    }
+  }
+
+  onSearchChanged = (searchText) => {
+    this.setState({searchText})
+    if (searchText === '') {
+      this.setState({searchMode: false})
+    } else {
+      this.setState({searchMode: true})
+      const matcher = new RegExp(searchText, 'i')
+      const searchResult = this.props.unitList
+        .filter(unit => !matcher || (unit.name && unit.name.match(matcher)))
+      this.setState({
+        searchResult: searchResult
+      })
+    }
+  }
+
+  render () {
+    const { show, onDismissed, inProgress, dispatch, unitList, item, isLoading } = this.props
+    const { searchText, searchMode, searchResult } = this.state
+    const units = searchMode ? searchResult : unitList
+    const actions =
+      <div>
+        <div className={'bondi-blue tc pa2'} onClick={() => dispatch(push(`/unit/new`))} >
+          <span className={units.length === 0 && 'dn'}> Add a new unit </span>
+        </div>
+      </div>
+    return (
+      <Dialog
+        title='Select Unit'
+        actions={actions}
+        titleStyle={customTitleStyle}
+        open={show}
+        autoDetectWindowHeight
+        autoScrollBodyContent
+        repositionOnUpdate
+        contentStyle={{
+          width: '90%',
+          height: '90%'
+        }}
+        bodyStyle={{
+          paddingBottom: '0',
+          borderTop: 'none',
+          borderBottom: 'none'
+        }}
+      >
+        {!inProgress && (
+          <button onClick={onDismissed}
+            className='button b--none bg-transparent absolute top-1 right-1'
+          >
+            <FontIcon className='material-icons' style={closeDialogButtonStyle}>close</FontIcon>
+          </button>
+        )}
+        <div>
+          <span className='ba b--moon-gray mv1 br1 pv1'>
+            <span className='ml3 absolute left-1' style={{marginTop: '0.8rem'}}>
+              <FontIcon className={'material-icons'} style={closeDialogButtonStyle}>
+                search
+              </FontIcon>
+            </span>
+            <TextField
+              hintText={<span className='ml1 '>Unit names </span>}
+              underlineShow={false}
+              fullWidth
+              value={searchText}
+              onChange={(evt) => this.onSearchChanged(evt.target.value)}
+              style={{paddingLeft: '2rem'}}
+              inputStyle={{padding: '0.3rem'}}
+            />
+          </span>
+        </div>
+        <div className='ba b--moon-gray pa2 br1' style={{minHeight: '380px'}}>
+          { isLoading ? (<Preloader />
```

Comment:
Use `<CircularProgress ... />` instead. The `<Preloader` is only meant to be used as a full page preloader as it takes up the entire height of the screen and will not be centered in the container